### PR TITLE
Improvement: Refactor task notifications and upgrade event conditions

### PR DESCRIPTION
**Pull request by abdullai-t**
####  Summary / Highlights

In this pull request, I refactored the task notification functionalities in our codebase and improved the event eligibility conditions for increased accuracy.
Key improvements include:

- Replaced the previous method of sending task notifications via Slack messages with a new send_slack_notification function. This helps streamline the reporting process of our task statuses.

- Modified the conditions used to determine the eligibility of certain events. With these changes, we hope to ensure the correct events are being tracked and reported.

- Introduced a new environment variable named SLACK_CELERY_TASKS_WEBHOOK_URL. This will help to better manage the settings of the Slack events.

#### Requirements (place an `x` in each `[ ]`)
* [x] I've read and understood the [Contributing Guidelines](/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've tested my changes manually.
* [ ] I've added unit tests to my PR.

##### Transparency ([Project board](https://github.com/orgs/massenergize/projects/7/views/2))
* [x] I've given my PR a meaningful title.
* [x] I linked this PR to the relevant issue.
* [ ] I moved the linked issue from the "Sprint backlog" to "In progress" when I started this.
* [ ] I moved the linked issue to "QA Verification" now that it is ready to merge.

### CODE REVIEW

File: src/task_queue/nudges/user_event_nudge.py [hunk 1]
```diff
@@ -130,13 +130,13 @@ def is_event_eligible(event, community_id, task=None):
         if freq:
             last_last_run = now - freq_to_delta.get(freq, relativedelta(days=0))
         
-        if settings.when_first_posted and event.published_at and last_last_run < event.published_at.date() <= now:
+        if settings.when_first_posted and event.published_at and (now - last_last_run).days < (
```
Comment (BradHN1) [inline src/task_queue/nudges/user_event_nudge.py:133]: Is this fixing a bug, where the notification rules are not working now?

File: src/task_queue/tasks.py [hunk 2]
```diff
@@ -32,21 +45,13 @@ def run_some_task(self, task_id):
                 try:
                     res = func(task)
                     task.status = "SUCCEEDED" if res else "FAILED"
-                    if IS_PROD:
-                        send_slack_message(SLACK_SUPER_ADMINS_WEBHOOK_URL, {
-                            "text": f"({ENV}): Task '{task.job_name}' was run on {today}. Status: {task.status} "})
+                    send_slack_notification(task.status, task.job_name, "Task ran successfully" if res else "Task failed")
```
Comment (BradHN1) [inline src/task_queue/tasks.py:48]: do not post about failed tasks that are not in PROD.

File: src/_main_/settings.py [hunk 3]
```diff
@@ -287,6 +287,7 @@
 POSTMARK_DOWNLOAD_SERVER_TOKEN = os.environ.get('POSTMARK_DOWNLOAD_SERVER_TOKEN')
 SLACK_COMMUNITY_ADMINS_WEBHOOK_URL = os.environ.get('SLACK_COMMUNITY_ADMINS_WEBHOOK_URL')
 SLACK_SUPER_ADMINS_WEBHOOK_URL = os.environ.get('SLACK_SUPER_ADMINS_WEBHOOK_URL')
+SLACK_CELERY_TASKS_WEBHOOK_URL = os.environ.get('SLACK_CELERY_TASKS_WEBHOOK_URL')
```
Comment (BradHN1) [inline src/_main_/settings.py:290]: we have one superadmin notification channel.  Please don't use a different one for this.

Comment (abdullai-t) [reply src/_main_/settings.py:290]: The super admin notifications channel is quite busy with various actions, events, testimonials, and other alerts, making it a bit challenging to quickly identify the cause of an alert at a glance. That's why I separated the Celery alerts into a different channel. Yesterday, @Opoku-Agyemang proposed some improvements to this feature, suggesting that we no longer need to use Slack as a logging platform for this purpose.

Comment (BradHN1) [reply src/_main_/settings.py:290]: There hasn't been a message in that channel in almost 2 months.  I prefer a single super admin alert Slack channel, which just important alerts that the super admin team need to take note of, not multiple channels.

Comment (abdullai-t) [reply src/_main_/settings.py:290]: I will do that

**Pull request closed without merging**